src/App.jsx

```diff
@@ -1,22 +1,33 @@
-import './App.css';
+import "./App.css";
 
-import AddTodoForm from './AddTodoForm';
-import TodoList from './TodoList';
-import { useState } from 'react';
+import { useEffect, useState } from "react";
+
+import AddTodoForm from "./AddTodoForm";
+import TodoList from "./TodoList";
+
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || []);
```

Comment:
Nit: As a good practice, store the local storage key in a variable, then use that when getting and setting from local storage. This will make the code more maintainable.